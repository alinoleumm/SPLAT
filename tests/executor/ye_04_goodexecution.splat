program


	
	printPoint (p1:Point1, num: Integer): void is
	
	begin
		print "point";
		print num;
		print "(";
		print p1.x;
		print ", ";
		print p1.y;
		print ")";
		print_line;
	end;
	
	record Point1 begin
		x: Integer;
		y: Integer;
	end;
	
	p1: Point1;
	p2: Point1;
begin
	p1 := Point1(10, 10);
	p2 := Point1(20, 20);
	
	print "printing points values before change:";
	print_line;
	printPoint(p1, 1);
	printPoint(p2, 2);
	
	p1 := p2;
	
	print "printing points values after change:";
	print_line;
	printPoint(p1, 1);
	printPoint(p2, 2);
end;