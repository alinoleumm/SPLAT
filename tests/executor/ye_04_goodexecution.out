printing points values before change:
point1(10, 10)
point2(20, 20)
printing points values after change:
point1(20, 20)
point2(20, 20)
